apps/docs/content/guides/auth/quickstarts/react.mdx

```diff
@@ -68,45 +66,101 @@ hideToc: true
             })
-
             const {
               data: { subscription },
             } = supabase.auth.onAuthStateChange((_event, session) => {
-              setSession(session)
+            setSession(session)
             })
+          return () => subscription.unsubscribe()
+        }, [])
+
+        async function signInWithEmail(e) {
+            e.preventDefault();
+            setLoading(true);
+            const { error } = await supabase.auth.signInWithPassword({ email, password });
+            if (error) alert(error.message);
+            setLoading(false);
+        }
 
-            return () => subscription.unsubscribe()
-          }, [])
-
-          if (!session) {
-            return (<Auth supabaseClient={supabase} appearance={{ theme: ThemeSupa }} />)
-          }
-          else {
-            return (<div>Logged in!</div>)
-          }
+        async function signUpWithEmail() {
+            setLoading(true);
+            const { data: { session }, error } = await supabase.auth.signUp({ email, password });
+            if (error) alert(error.message);
+            if (!session) alert('Please check your inbox for email verification!');
+            setLoading(false);
         }
+
+        return (
+            <div>
+                {session ? (
+                    <div> Logged in </div>
+                ) : (
+                <form onSubmit={signInWithEmail}>
+                    <div>
+                        <label htmlFor="email">
+                        Email:
+                        <input
+                            id="email"
+                            name="email"
+                            type="email"
+                            value={email}
+                            autocomplete="email"
+                            onChange={(e) => setEmail(e.target.value)}
+                            placeholder="email@address.com"
+                        />
+                        </label>
+                    </div>
+                    <div>
+                        <label htmlFor="password">
+                        Password:
+                        <input
+                            id="password"
+                            name="password"
+                            type="password"
+                            value={password}
+                            onChange={(e) => setPassword(e.target.value)}
+                            placeholder="Password"
+                        />
+                        </label>
+                    </div>
+                    <button
+                        type="submit"
+                        disabled={loading}
+                        >
+                    Sign in
+                    </button>
+                    <button
+                        type="button"
+                        onClick={signUpWithEmail}
+                        disabled={loading}
+                        >
+                    Sign up
+                    </button>
```

Comment:
you have 2 submit buttons inside a form

Reply:
button is default a submit type.

it's also not clear what would happen if you submitted the form with keyboard (enter) it will probably default to sign in.